R/feature_helpers.R

```diff
@@ -166,3 +183,88 @@ validExtension <- function(extension, cageFiveUTRs) {
     stop("if extension is not 0, then cageFiveUTRs must be defined")
   }
 }
+
+#' Find proportion of reads per position in ORF TIS window
+#'
+#' Proportion defined as:
+#' average count per position in -20,20 normalized by counts per gene
+#'
+#' This pattern can be averaged on CDS's, to find other ORFs.
+#' When detecting new ORFs, this CDS average can be used as a template.
+#' @param grl a \code{\link{GRangesList}} object
+#'  with usually new ORFs, but can also be
+#'  either leaders, cds', 3'UTRs.
+#' @param tx a GrangesList of transcripts covering grl.
+#' @param footprints ribo seq reads as GAlignment, GRanges
+#'  or GRangesList object.
+#' @param onlyProportion a logical (FALSE), return whole data.frame or only
+#'  proportions
+#' @param average a logical (FALSE), take average over coverage in all grl ?
+#' @param pShifted a logical (TRUE), are riboseq reads p-shifted?
+#' @param keep.names a logical(FALSE), only applies when onlyProportion
+#'  is TRUE.
+#' @return a data.frame with lengths by coverage / vector of proportions
+#' @importFrom BiocGenerics Reduce
+#'
+riboTISCoverageProportion <- function(grl, tx, footprints,
+                                      onlyProportion = FALSE, average = FALSE,
+                                        pShifted = TRUE, keep.names = FALSE){
```

Comment:
formatting, space between ) {, you could move outside the upStart and downStop params to get more flexible function